homework/index.js

```diff
@@ -1,53 +1,143 @@
+    });
+    createTable(table, 'Description: ', repo.description);
+    createTable(table, 'Fork count: ', repo.forks);
+    createTable(table, 'Created on: ', changeDateFormat(repo.created_at));
+    createTable(table, 'Updated on: ', changeDateFormat(repo.updated_at));
+  }
+
+  function contributorsSection(url, parent) {
+    fetchJSON(url)
+      .then(repos => {
+        repos.forEach(repo => {
+          const li = createAndAppend('li', parent, { class: 'rightList' });
+          createAndAppend('img', li, {
+            class: 'image',
+            src: repo.avatar_url,
+          });
+          createAndAppend('a', li, {
+            text: repo.login,
+            href: repo.html_url,
+            target: '_blank',
+            class: 'cont-link',
+          });
+          createAndAppend('p', li, {
+            class: 'contributions',
+            text: repo.contributions,
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', parent, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
+  }
+
+  function repoSelect(item, repoItem, parent) {
+    const ul = createAndAppend('ul', parent);
+    const repo = repoItem[item];
+    renderRepoDetails(repo, ul);
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      text: 'HYF Repositories',
+    });
+    const mainElem = createAndAppend('main', root, { class: 'main-container' });
+    const repoContainer = createAndAppend('section', mainElem, {
+      class: 'repo-container',
+    });
+    const ul = createAndAppend('ul', repoContainer);
+    const rightSection = createAndAppend('section', mainElem, {
+      class: 'contributors-container',
+    });
+    const p = createAndAppend('p', rightSection, { class: 'right-title' });
+    p.innerHTML = 'Contributions';
+    const rightUl = createAndAppend('ul', rightSection, { class: 'right-ul' });
+    const select = createAndAppend('select', header, { class: 'test' });
+    fetchJSON(url)
+      .then(repoItem => {
+        renderRepoDetails(repoItem[17], ul);
+        const startItem = repoItem[17].contributors_url;
+        contributorsSection(startItem, rightUl);
+        repoItem
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .forEach((repo, index) => {
+            createAndAppend('option', select, {
+              text: repo.name,
+              value: index,
+            });
+          });
+        select.addEventListener('change', () => {
+          repoContainer.innerHTML = '';
+          rightUl.innerHTML = '';
+          const selectInput = select.value;
+          repoSelect(selectInput, repoItem, repoContainer);
+          const contUrl = repoItem[selectInput].contributors_url;
+          contributorsSection(contUrl, rightUl);
+        });
+      })
```

Comment:
I'm repeating the same comment I gave  @ahmthsrvsmz:

Basically you need to execute the same code to render a repository in two different places:

1. Once, at application start-up, using the repository at index zero.
2. Inside the change event handler, using the index of the selected repo.

When you detect a repeated code like this you should immediately consider writing a function to remove this duplication. Can try and write such a function?

Reply:
Right. I am gonna revisit the lines for DRY.

Reply:
Your code is still not DRY (enough). If you carefully read my hint in the comment below, you can write a function that handles both the rendering of the first repo at startup and the rendering of the selected repo when the user changes the selection in the select box. When the user changes the selection, you need to wipe out the details of the previous selection (i.e. clearing out `.innerHTML`). However, although redundant because the containers are initially empty, there it is no problem if you also do that at start-up. With that in mind, you could write a function likes this:

```js
function render(repo, repoContainer, rightUl) {
  repoContainer.innerHTML = '';
  rightUl.innerHTML = '';
  renderRepoDetails(repo, repoContainer);
  getContributorsData(repo.contributors_url, rightUl);
}
```

In your main function, you could use it like this:

```js
...
render(repositories[0], repoContainer, rightUl);
select.addEventListener('change', () => {
  render(repositories[select.value], repoContainer, rightUl);
});
...
```

The function `renderSelect` is no longer needed.